app/controllers/users_controller.rb

```diff
@@ -4,8 +4,25 @@ class UsersController < ApplicationController
 
 
   def index
-    @users = User.all.order(last_sign_in_at: :asc)
-      .includes(:membership_applications)
+    @q = User.ransack(params[:q])
+    @users = @q.result.includes(:membership_applications)
+
+    @filter_are_members = params.include?(:are_members)
+    @filter_are_not_members = params.include?(:are_not_members)
+
+    if not (@filter_are_members || @filter_are_not_members)
+      permitted = params.permit(q: [:email_cont, :s])
+      redirect_to controller: :users, action: :index,
+        params: { are_members: 1, are_not_members: 1, q: permitted[:q] }
+    end
+
+    unless params[:are_members] && params[:are_not_members]
+      %w(are_members are_not_members).each do | filter |
+        @users = @users.send(filter) if params[filter]
+      end
+    end
```

Comment:
This is a matter of preference, but I am not a fan of redirecting from an action to the exact same action, if it can avoided.  I played around with the logic in lines 10-23 and used this instead:
```
@filter_are_members = params.include?(:are_members)
@filter_are_not_members = params.include?(:are_not_members)

one_filter_set = (@filter_are_members && !@filter_are_not_members) ||
                 (@filter_are_not_members && !@filter_are_members)

if one_filter_set
  %w(are_members are_not_members).each do | filter |
    @users = @users.send(filter) if params[filter]
  end
elsif !(@filter_are_members && @filter_are_not_members)

  # If both filters are missing from params, what has occurred is either of:
  # 1. Page is initially loading, or,
  # 2. User has deselected both filters
  # For case 1, we show all users by default, so show the checkboxes as checked
  # For case 2, normal processing would show no results, which is not useful,
  #             so we reselect both checkboxes (may want to add flash message explaining this to user)

  @filter_are_members = true
  @filter_are_not_members = true
end
```

Either way is OK for me.

And, either way, this logic is a bit tricky to understand quickly, so strongly recommend comments where appropriate (as in my example).